### PR TITLE
SearchTakeBestJob, SearchRemoveLabelJob, SearchBeamJoinScoresJob

### PR DESCRIPTION
**Pull request by albertz**
Three jobs which I needed for my recog pipeline.

I can make three separate PRs if you prefer, but on the other side, all of them are quite trivial, so maybe we can directly do it in one go? Just take a short look.

### CODE REVIEW

**Comment by JackTemaki** [comment]
As they are all related all in one PR is fine! Will check after lunch.

**Review by michelwi — APPROVED** [review]
Looks good to me. Only one optional comment for improvement.

File: returnn/search.py [hunk 1]
```diff
@@ -478,3 +478,133 @@ def run(self):
...
+                out.write("%r: %r,\n" % (seq_tag, best_entry))
+            out.write("}\n")
+
+
+class SearchRemoveLabelJob(Job):
+    """
+    Remove some labels from the search output, e.g. "<blank>".
+    """
+
+    def __init__(self, search_py_output: tk.Path, *, remove_label: str):
```
[the remark this answers is not in the document]
Comment (albertz) [reply returnn/search.py]: Implemented that (but `Union[str, Set[str]]` instead).

**Review by michelwi — APPROVED** [review]

**Review by michelwi — APPROVED** [review]

**Review by JackTemaki — APPROVED** [review]
Willi already mentioned the only issue, looks good.

We might run into trouble if the search output is large (e.g. you search a whole training corpus), but we can adapt the jobs then.

**Pull request merged**